### ISSUE TITLE
Improve TypeScript support in Deno

### ISSUE THREAD

**Issue by memark** [selfrefactor/rambda#612]
TypeScript support in Deno for Rambda currently lacking a bit. Here is a suggestion of steps to fix that:

1. Copy the _index.d.ts_ file to _dist/_
2. Add `/// <reference types="./index.d.ts" />` to the top of the _dist/rambda.esm.js_ file
3. Change the _deno.land_ binding to publish the entire repo (this is the way it's usually done). Right now only the _src/_ folder is published, which forces you to import a separate js file for every function you want to use (so no one will use it).
4. Change the README to say _For deno: import {compose, add} from 'https://deno.land/x/rambda/dist/rambda.esm.js'_

And everything will be splendour!

I'd be happy to send PRs or otherwise help out with this.

**Comment by selfrefactor** [comment]
No need for PR as your suggestions are well described. Still, I have a few questions:

- Are you sure that adding `/// <reference types="./index.d.ts" />` won't have negative effect for non-Deno projects? 
- Any suggestion or reference to help me in `Change the deno.land binding to publish the entire repo `?

**Comment by memark** [comment]
> Are you sure that adding /// <reference types="./index.d.ts" /> won't have negative effect for non-Deno projects?

I did some testing locally. Create a local Node app, installed Rambda, added the `/// <reference...` line to the relevant file in `node_modules...` and imported Rambda into the application. I found no problems.

> Any suggestion or reference to help me in Change the deno.land binding to publish the entire repo?

Not sure how to do this. I did notice that when trying out the GUI for publishing a new package on deno.land/x, a field for selecting a subdirectory is presented.

<img width="677" alt="image" src="https://user-images.githubusercontent.com/318504/151656005-81d4444c-3680-4d9b-8dcf-dfe61c9763a0.png">

Perhaps this can be changed afterwards as well?
